src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -1007,13 +1008,29 @@ private ICrossChainTransfer[] GetTransfersByStatus(CrossChainTransferStatus[] st
         }
 
         /// <inheritdoc />
-        public Task<Dictionary<uint256, Transaction>> GetTransactionsByStatusAsync(CrossChainTransferStatus status, bool sort = false)
+        public ICrossChainTransfer[] GetTransfersByStatus(CrossChainTransferStatus[] statuses, bool sort = false)
         {
-            return Task.Run(() =>
-            {
-                ICrossChainTransfer[] res = this.GetTransfersByStatus(new[] { status }, sort);
-                return res.Where(t => t.PartialTransaction != null).ToDictionary(t => t.DepositTransactionId, t => t.PartialTransaction);
-            });
+            return this.GetTransfersByStatus(statuses, sort, false);
+        }
+
+        /// <inheritdoc />
+        public ICrossChainTransfer[] QueryTransfersByStatus(CrossChainTransferStatus[] statuses)
```

Comment:
We now have these 2 methods which don't do validation immediately - they just query the current state.

This is more useful for the console data.

Reply:
What about locking? any access to `depositsIdsByStatus` should be locked I would imagine.

Reply:
The only that had to happen is to make this method public:
`private ICrossChainTransfer[] GetTransfersByStatus(CrossChainTransferStatus[] statuses, bool sort = false, bool validate = true)`
We want to force the user to make an explicit decision to turn validation off, rather than have two methods where its not clear whether validation occurs or not.

Reply:
It probably make sense to not validate when the method is called `QueryXXX`